xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -1657,6 +1678,53 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
   return true;
 }
 
+bool CAMLCodec::OpenAmlVideo(const CDVDStreamInfo &hints)
+{
+  PosixFilePtr amlVideoFile = std::make_shared<PosixFile>();
```

Comment:
Yes, right, shared_ptr is used here only for auto-deletion. I took PosixFile class from my ION decoder PR, there was more complex initialization, so I used shared pointer there to avoid gotos or a lot of repeating calls to cleanup parts already initialized in a case of failure.